src/format_source.sh

```diff
@@ -6,9 +6,7 @@
 #
 
 # Get the directory containing this script
-pushd $(dirname $0) > /dev/null
-base=$(pwd -P)
-popd > /dev/null
+base="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
```

Comment:
This is just a stylistic choice.  I think the separate use of `pushd` and `popd` are more clear to read.  However the `format_source.sh` script has already changed in toast3, so I'll just keep it as the original in that branch.

Reply:
Admittedly this is something I picked up from https://stackoverflow.com/a/246128 in the past, which has evolved a bit.

I didn't think much when I changed that line, but thinking on why I changed it, probably it is because those statements requires more mental parsing as it is more stateful. Since those sub-shell is necessary anyway, getting an output from a subshell directly is less stateful (it is not changing anything in the current shell.)

Agree in the end it is just a stylistic choice as it works none the less!